quantum/dip_switch.c

```diff
@@ -34,7 +34,21 @@
 
 #ifdef DIP_SWITCH_PINS
 #    define NUMBER_OF_DIP_SWITCHES (sizeof(dip_switch_pad) / sizeof(pin_t))
+#    ifndef DIP_SWITCH_INVERT
+#       define DIP_SWITCH_INVERT {0}
+#    endif
 static pin_t dip_switch_pad[] = DIP_SWITCH_PINS;
+static uint8_t dip_switch_invert[NUMBER_OF_DIP_SWITCHES] = DIP_SWITCH_INVERT;
+
+#    if defined(SPLIT_KEYBOARD) && defined(DIP_SWITCH_PINS_RIGHT)
+#       ifndef DIP_SWITCH_INVERT_RIGHT
+#           define DIP_SWITCH_INVERT_RIGHT DIP_SWITCH_INVERT
+#       endif
+static pin_t dip_switch_pad_right[] = DIP_SWITCH_PINS_RIGHT;
+static uint8_t dip_switch_invert_right[NUMBER_OF_DIP_SWITCHES] = DIP_SWITCH_INVERT_RIGHT;
```

Comment:
The code that actually uses `dip_switch_invert_right` is inside `#if defined(DIP_SWITCH_INVERT_RIGHT)`; the same logic should be used here instead of defining the variable in any case.